src/main/java/racingcar/domain/car/Cars.java

```diff
@@ -0,0 +1,57 @@
+package racingcar.domain.car;
+
+import racingcar.strategy.MoveStrategy;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Cars {
+    private static final String DUPLICATED_NAME_EXCEPTION_MESSAGE = "중복되는 이름의 차를 입력하실 수 없습니다.";
+    private static final String RESULT_MESSAGE_DELIMITER = "\n";
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = new ArrayList<>(cars);
+        validate();
+    }
+
+    private void validate() {
+        cars.forEach(car -> {
+            long numberOfSameNameCars = cars.stream()
+                    .filter(car::equals)
+                    .count();
+
+            if (isMoreThanTwo(numberOfSameNameCars)) {
+                throw new IllegalArgumentException(DUPLICATED_NAME_EXCEPTION_MESSAGE);
+            }
+        });
```

Comment:
```suggestion
        if (cars.size() != new HashSet<>(cars).size()) {
            throw new IllegalArgumentException(DUPLICATED_NAME_EXCEPTION_MESSAGE);
        }
        });
```

중복에 관련된 부분은 Set을 사용하여 처리할 수도 있습니다 🙂

Reply:
신박한 방법입니다... 😮